tools/incubator.js

```diff
@@ -209,6 +209,25 @@ const Incubator = new Lang.Class({
             basin_manifest['content'].push(this._import_video(asset));
         });
 
+
+        /* Once we're done, look over all the content in the manifest and
+         * set the parent key on each metadata struct. */
+        let manifest_table = {};
+        basin_manifest.content.filter(metadata => metadata['@id'])
+                              .forEach(metadata =>
+                                  manifest_table[metadata['@id']] = metadata
+                              );
+
+        /* Having to use indexOf to check if the metadata has resources is not
+         * ideal, but trying to access the prop directly in filter raises
+         * a warning about undefined property use. */
+        basin_manifest.content.filter(metadata => Object.keys(metadata).indexOf('resources') !== -1)
```

Comment:
Can't you use `metadata.hasOwnProperty()`?

Reply:
That would work - lets do that.